Packs/Active_Directory_Query/Scripts/IAMInitADUser/IAMInitADUser.py

```diff
@@ -1,163 +1,220 @@
-        outputs['success'] = True
+    Args:
+        file_name (str): The name of the file to find.
 
-    except Exception as e:
-        outputs['success'] = False
-        outputs['errorDetails'] = str(e)
-        err = str(e)
+    Returns:
+        str: The entry ID of the file.
+    """
+    file_entries = demisto.context().get('File', [])
 
-    try:
-        send_mail_outputs = send_email(display_name, username, user_email, err,
-                                       to_email, password, inc_id, email_subject, email_body,
-                                       zip_protect_with_password)
+    if isinstance(file_entries, dict):  # In case of a single entry
+        file_entries = [file_entries]
 
-        if is_error(send_mail_outputs):
-            raise Exception(f'An error occurred while trying to send mail. Error is:\n{get_error(send_mail_outputs)}')
-        outputs['sentMail'] = True
+    for item in file_entries:
+        if item['Name'] == file_name:
+            return item['EntryID']
 
-    except Exception as e:
-        outputs['sentMail'] = False
-        outputs['sendMailError'] = str(e)
+    raise DemistoException(f"Could not find file '{file_name}' in the context.")
+
+
+def generate_password(password_generation_script: str, min_lcase: str, max_lcase: str, min_ucase: str, max_ucase: str,
+                      min_digits: str, max_digits: str, min_symbols: str, max_symbols: str) -> str:
+    """
+    Generate a random password using a script.
+
+    Args:
+        password_generation_script (str): The name of the script to use for generating the password.
+        min_lcase (str): The minimum number of lowercase characters to use in the password.
+        max_lcase (str): The maximum number of lowercase characters to use in the password.
+        min_ucase (str): The minimum number of uppercase characters to use in the password.
+        max_ucase (str): The maximum number of uppercase characters to use in the password.
+        min_digits (str): The minimum number of digits to use in the password.
+        max_digits (str): The maximum number of digits to use in the password.
+        min_symbols (str): The minimum number of symbols to use in the password.
+        max_symbols (str): The maximum number of symbols to use in the password.
+
+    Returns:
+        str: The generated password.
+    """
+    # Generate a random password
+    if password_generation_script == DEFAULT_PWD_GENERATION_SCRIPT:
+        script_params = {
+            "min_lcase": min_lcase,
+            "max_lcase": max_lcase,
+            "min_ucase": min_ucase,
+            "max_ucase": max_ucase,
+            "min_digits": min_digits,
+            "max_digits": max_digits,
+            "min_symbols": min_symbols,
+            "max_symbols": max_symbols
+        }
+
+        demisto.debug(f'Executing {password_generation_script} with the following arguments: {script_params}')
```

Comment:
please remove. if the args would be of interest, we'll ask to run it in debug-mode, otherwise no need to print this to the logs.